### PR TITLE
Fix type annotation for mypy check of examples

### PR DESCRIPTION
**Pull request by asi1024**
Depends on https://github.com/pfnet/pytorch-pfn-extras/pull/348.

### CODE REVIEW

File: example/mnist_custom_logic.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def forward(self, x):
         return F.log_softmax(x, dim=1)
 
 
-class CustomLogic(ppe.handler.Logic):
+class CustomLogic(ppe.handler.Logic):  # type: ignore[misc]
```
Comment (kmaehashi) [inline example/mnist_custom_logic.py:31]: Can we avoid this?

File: example/mnist_ddp.py [hunk 2]
```diff
@@ -141,11 +141,11 @@ def main():
             transforms.Normalize((0.1307,), (0.3081,)),
         ]))
 
-    train_sampler = torch.utils.data.DistributedSampler(
+    train_sampler = torch.utils.data.DistributedSampler(  # type: ignore[var-annotated]
```
Comment (kmaehashi) [inline example/mnist_ddp.py:144]: Can we avoid this?

File: example/mnist_ddp.py [hunk 3]
```diff
@@ -48,7 +48,7 @@ def test(args, model, device, data, target):
         averages
     """
     model.eval()
-    test_loss = 0
+    test_loss = 0.
```
Comment (kmaehashi) [inline example/mnist_ddp.py:51]:
```suggestion
    test_loss = 0.0
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]

**Comment by kmaehashi** [comment]
Could you fix flake8 errors?

**Comment by asi1024** [comment]
PTAL!

File: pytorch_pfn_extras/training/manager.py [hunk 4]
```diff
@@ -98,19 +100,19 @@ def __init__(
...
-        if not isinstance(models, dict):
+        if not isinstance(models, collections.abc.Mapping):
             if not isinstance(models, torch.nn.Module):
                 raise ValueError(
                     'model must be an instance of dict or toch.nn.Module')
             self._models = {'main': models}
         else:
-            self._models = models
-        if not isinstance(optimizers, dict):
+            self._models = dict(models)
```
Comment (kmaehashi) [inline pytorch_pfn_extras/training/manager.py:109]: Can this be avoided?

File: pytorch_pfn_extras/training/manager.py [hunk 5]
```diff
@@ -98,19 +100,19 @@ def __init__(
...
+        if not isinstance(optimizers, collections.abc.Mapping):
             # TODO(ecastill) Optimizer type is not checked because of tests
             # using mocks and other classes
             self._optimizers = {'main': optimizers}
         else:
-            self._optimizers = optimizers
+            self._optimizers = dict(optimizers)
```
Comment (kmaehashi) [inline pytorch_pfn_extras/training/manager.py:115]: ditto

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**